data/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -486,4 +490,22 @@ public void testPrimitiveTypeDefaultValues(Type.PrimitiveType type, Object defau
 
     writeAndValidate(writeSchema, readSchema);
   }
+
+  @Test
+  public void testWriteNullValueForRequiredType() {
+    Assumptions.assumeThat(supportsDefaultValues()).isTrue();
```

Comment:
where is `supportsDefaultValues()` defined?

Reply:
It's defined in this class and overridden in the subclasses.

Reply:
I've replaced it with `supportsRequiredValues()`. There is actually only one class, `org.apache.iceberg.data.orc.TestGenericData` which does not support it.